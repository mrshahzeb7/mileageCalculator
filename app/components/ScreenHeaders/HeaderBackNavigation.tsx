import React from 'react';
import {StyleSheet, Text, View, ViewStyle} from 'react-native';
import {useSafeAreaInsets} from 'react-native-safe-area-context';
import {icons} from '../../assets';
import {colors, fontSize, iconSizes, spacing, styleGlobals} from '../../theme';
import {responsiveHeight} from '../../utils';
import {Icon} from '../Icon/Icon';

interface Props {
  onLeftPress?: () => void;
  title: string;
  RightAction?: any;
}

export const HeaderBackNavigation: React.FC<Props> = ({
  onLeftPress,
  title,
  RightAction,
}) => {
  const padding: ViewStyle = {
    paddingTop: useSafeAreaInsets().top,
  };

  return (
    <View style={[styles.container, padding]}>
      <View style={styles.wrapper}>
        <View style={styles.leftWrapper}>
          <Icon
            testID="HeaderBackButton"
            onPress={onLeftPress}
            size={iconSizes.small}
            color={colors.light}
            icon={icons.Back}
          />

          <Text style={styles.title}>{title}</Text>
        </View>

        {!!RightAction && RightAction}
      </View>
    </View>
  );
};

const $row: ViewStyle = {
  flexDirection: 'row',
  alignItems: 'center',
};
// styles
const styles = StyleSheet.create({
  container: {
    height: responsiveHeight(12),

    paddingHorizontal: spacing.horizontal.medium,
    justifyContent: 'center',
    ...styleGlobals.lowElevation,
    backgroundColor: colors.primary,
  },
  wrapper: {
    ...$row,
    justifyContent: 'space-between',
  },
  leftWrapper: {
    ...$row,
  },
  title: {
    color: colors.text,
    fontSize: fontSize.large,
    marginLeft: spacing.horizontal.medium,
  },
});
