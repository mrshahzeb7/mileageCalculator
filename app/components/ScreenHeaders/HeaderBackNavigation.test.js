import 'react-native';
import React from 'react';
import {HeaderBackNavigation} from './HeaderBackNavigation';
import renderer from 'react-test-renderer';
import {
  SafeAreaProvider,
  initialWindowMetrics,
} from 'react-native-safe-area-context';

const Parent = () => {
  return (
    <SafeAreaProvider initialMetrics={initialWindowMetrics}>
      <HeaderBackNavigation />
    </SafeAreaProvider>
  );
};

describe('HeaderBackNavigation', () => {
  it(' HeaderBackNavigation renders correctly', () => {
    const tree = renderer.create(<Parent />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
