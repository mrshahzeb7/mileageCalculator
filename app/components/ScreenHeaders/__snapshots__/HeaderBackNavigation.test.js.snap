// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBackNavigation  HeaderBackNavigation renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
/>
`;
